### DISCUSSION TITLE
Use Async functions and callbacks in background

### DISCUSSION THREAD

**Discussion by michaelessiet** [wxt-dev/wxt#389] [Q&A] [answered]
Hi, I'm currently working on this extension that needs to call an async function and a couple async callbacks in the background service worker. The expected behavior is for it to run all functions like any other javascript environment would, however the current behavior has the background.js file not running at all, it doesn't throw an error either.

Help on this would be much appreciated. BTW thanks for the awesome framework, it saved me a lot of time and configuration.

**Comment by aklinker1** [comment, marked as answer]
Hey, thanks for asking. Not sure what's going wrong without seeing your code, but here's an example extension that makes an network request in the background.

Note that the background's main function cannot be async, because you need to register all extension listeners synchronously when the service worker starts up. So to call an async API, like `fetch`, you have to use `.then` or a callback to handle the result.

[wxt-async-background-example.zip](https://github.com/wxt-dev/wxt/files/14100919/wxt-async-background-example.zip)

https://github.com/wxt-dev/wxt/assets/10101283/eabfd998-bc2b-4e36-ad0a-cc4d26ad80f3

**Reply by michaelessiet** [reply]
Oh never mind, the issue seems to have been originating from the StorageService. It might have been because I was importing `crypto-browserify`. Thank you for showing me that it async functions work.
